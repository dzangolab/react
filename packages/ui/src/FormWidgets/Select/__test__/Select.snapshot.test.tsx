import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";

import { Select } from "..";

const options = [
  { value: "np", label: "Nepali" },
  { value: "en", label: "English" },
];

describe("Select Component", () => {
  test("should render correctly with single selection", () => {
    const { container } = render(
      <Select
        name="single-select"
        options={options}
        value="en"
        onChange={() => {}}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render correctly with multiple selection", () => {
    const { container } = render(
      <Select
        name="multi-select"
        options={options}
        value={["np", "en"]}
        multiple
        onChange={() => {}}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render correctly with a placeholder", () => {
    const { container } = render(
      <Select
        name="placeholder-select"
        options={options}
        value=""
        placeholder="Select a language"
        onChange={() => {}}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render correctly with an error message", () => {
    const { container } = render(
      <Select
        name="error-select"
        options={options}
        value="en"
        hasError
        errorMessage="This field is required"
        onChange={() => {}}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render correctly with helper text", () => {
    const { container } = render(
      <Select
        name="helper-text-select"
        options={options}
        value="np"
        helperText="Please select an option"
        onChange={() => {}}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render correctly when disabled", () => {
    const { container } = render(
      <Select
        name="disabled-select"
        options={options}
        value="en"
        disabled
        onChange={() => {}}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render correctly when auto-select single option is enabled", () => {
    const { container } = render(
      <Select
        name="auto-select-single"
        options={options}
        value="np"
        autoSelectSingleOption
        onChange={() => {}}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render correctly when hide if single option is enabled", () => {
    const { container } = render(
      <Select
        name="hide-if-single"
        options={options}
        value="np"
        hideIfSingleOption
        onChange={() => {}}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
