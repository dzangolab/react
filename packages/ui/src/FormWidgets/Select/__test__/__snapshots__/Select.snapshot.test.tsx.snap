// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Select Component > should render correctly when auto-select single option is enabled 1`] = `
<div>
  <div
    class="field"
  >
    <div
      class="select"
    >
      <div
        class="label-container"
        tabindex="-1"
      >
        <div
          class="field debounced-input"
        >
          <input
            class="input-field "
            id=""
            name=""
            tabindex="0"
            type="text"
            value="Nepali"
          />
        </div>
        <span
          class="menu-trigger"
        >
          <i
            class="pi pi-chevron-down"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select Component > should render correctly when disabled 1`] = `
<div>
  <div
    class="field"
  >
    <div
      class="select"
    >
      <div
        class="label-container disabled"
        tabindex="-1"
      >
        <div
          class="field debounced-input"
        >
          <input
            class="input-field "
            id=""
            name=""
            tabindex="0"
            type="text"
            value="English"
          />
        </div>
        <span
          class="menu-trigger"
        >
          <i
            class="pi pi-chevron-down"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select Component > should render correctly when hide if single option is enabled 1`] = `
<div>
  <div
    class="field"
  >
    <div
      class="select"
    >
      <div
        class="label-container"
        tabindex="-1"
      >
        <div
          class="field debounced-input"
        >
          <input
            class="input-field "
            id=""
            name=""
            tabindex="0"
            type="text"
            value="Nepali"
          />
        </div>
        <span
          class="menu-trigger"
        >
          <i
            class="pi pi-chevron-down"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select Component > should render correctly with a placeholder 1`] = `
<div>
  <div
    class="field"
  >
    <div
      class="select"
    >
      <div
        class="label-container"
        tabindex="-1"
      >
        <div
          class="field debounced-input"
        >
          <input
            class="input-field "
            id=""
            name=""
            placeholder="Select a language"
            tabindex="0"
            type="text"
            value=""
          />
        </div>
        <span
          class="menu-trigger"
        >
          <i
            class="pi pi-chevron-down"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select Component > should render correctly with an error message 1`] = `
<div>
  <div
    class="field"
  >
    <div
      class="select"
    >
      <div
        aria-invalid="true"
        class="label-container"
        tabindex="-1"
      >
        <div
          class="field debounced-input"
        >
          <input
            class="input-field "
            id=""
            name=""
            tabindex="0"
            type="text"
            value="English"
          />
        </div>
        <span
          class="menu-trigger"
        >
          <i
            class="pi pi-chevron-down"
          />
        </span>
      </div>
    </div>
    <span
      class="error-message"
    >
      This field is required
    </span>
  </div>
</div>
`;

exports[`Select Component > should render correctly with helper text 1`] = `
<div>
  <div
    class="field"
  >
    <div
      class="select"
    >
      <div
        class="label-container"
        tabindex="-1"
      >
        <div
          class="field debounced-input"
        >
          <input
            class="input-field "
            id=""
            name=""
            tabindex="0"
            type="text"
            value="Nepali"
          />
        </div>
        <span
          class="menu-trigger"
        >
          <i
            class="pi pi-chevron-down"
          />
        </span>
      </div>
    </div>
    <span
      class="helper-text"
    >
      Please select an option
    </span>
  </div>
</div>
`;

exports[`Select Component > should render correctly with multiple selection 1`] = `
<div>
  <div
    class="field"
  >
    <div
      class="select"
    >
      <div
        class="label-container"
        tabindex="-1"
      >
        <div
          class="field debounced-input"
        >
          <input
            class="input-field "
            id=""
            name=""
            tabindex="0"
            type="text"
            value="Nepali, English"
          />
        </div>
        <span
          class="menu-trigger"
        >
          <i
            class="pi pi-chevron-down"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select Component > should render correctly with search enabled 1`] = `
<div>
  <div
    class="field"
  >
    <div
      class="select"
    >
      <div
        class="label-container"
        tabindex="-1"
      >
        <div
          class="field debounced-input"
        >
          <input
            class="input-field "
            id=""
            name=""
            tabindex="0"
            type="text"
            value="Nepali"
          />
        </div>
        <span
          class="menu-trigger"
        >
          <i
            class="pi pi-chevron-down"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select Component > should render correctly with single selection 1`] = `
<div>
  <div
    class="field"
  >
    <div
      class="select"
    >
      <div
        class="label-container"
        tabindex="-1"
      >
        <div
          class="field debounced-input"
        >
          <input
            class="input-field "
            id=""
            name=""
            tabindex="0"
            type="text"
            value="English"
          />
        </div>
        <span
          class="menu-trigger"
        >
          <i
            class="pi pi-chevron-down"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
